pkg/externalresource/external_resource.go

```diff
@@ -0,0 +1,65 @@
+package externalresource
+
+import (
+	b64 "encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+// ExternalResources represents the map of external resources at a manifest
+type ExternalResourceSection map[string]*ExternalResource
+
+// ExternalResource represents information on an external resource
+type ExternalResource struct {
+	Icon      string
+	Notes     Notes
+	Endpoints []ExternalEndpoint
+}
+
+// Notes represents information about the location and content of the external resource markdown
+type Notes struct {
+	Path     string
+	Markdown string // base64 encoded content of the path
+}
+
+// ExternalEndpoint represents information about an endpoint
+type ExternalEndpoint struct {
+	Name string
+	Url  string
+}
+
+// ERFilesystemManager represents ExternalResource information with the filesystem inyected
+type ERFilesystemManager struct {
```

Comment:
This is the only thing that annoys me... We create an struct to modify one of their fields. I think it can be better if we return externalresource modified. WDYT?

Reply:
In my opinion I think that the name of the function helps not to create confusion, it simply loads the content of the markdown. Receiving as parameter a struct and returning it modified or creating a new one I believe that leaves us in a similar situation and even you are left with a more verbose function declaration. But that's my opinion, I have no strong opinion for or against it

Reply:
What if is `ExternalResource` the one that has the afero dependency? That way we don't have to create another an intermediate struct. `externalResource.LoadMarkdownContent(path string)`. 

We could even not add the afero dependency as a field and initialise it inside the `externalResource.LoadMarkdownContent`.

Reply:
in most of the use of this struct this property would not be used, I believe that it is something foreign to the information that must store the `ExternalResource` struct. If we instantiate it inside the function we can not use the injection of this variable for its testing